components/cosmetic_filters/renderer/cosmetic_filters_js_handler.cc

```diff
@@ -297,31 +299,30 @@ void CosmeticFiltersJSHandler::OnRemoteDisconnect() {
   EnsureConnected();
 }
 
-void CosmeticFiltersJSHandler::ProcessURL(const GURL& url,
-                                          base::OnceClosure callback) {
+bool CosmeticFiltersJSHandler::ProcessURL(const GURL& url) {
   resources_dict_.reset();
   url_ = url;
   // Trivially, don't make exceptions for malformed URLs.
   if (!EnsureConnected() || url_.is_empty() || !url_.is_valid())
-    return;
+    return false;
 
-  cosmetic_filters_resources_->UrlCosmeticResources(
-      url_.spec(),
-      base::BindOnce(&CosmeticFiltersJSHandler::OnUrlCosmeticResources,
-                     base::Unretained(this), std::move(callback)));
-}
+  auto* content_settings =
+      static_cast<content_settings::BraveContentSettingsAgentImpl*>(
+          content_settings::ContentSettingsAgentImpl::Get(render_frame_));
 
-void CosmeticFiltersJSHandler::OnUrlCosmeticResources(
-    base::OnceClosure callback,
-    bool enabled,
-    bool first_party_enabled,
-    base::Value result) {
-  if (!enabled || !EnsureConnected())
-    return;
-  enabled_1st_party_cf_ = first_party_enabled;
```

Comment:
not a new issue with this PR, but this should probably be reset to `false` when calling `ProcessURL`?

Reply:
By the time `ProcessURL` returns, it'll either be set correctly or `ProcessURL` will return `false` to signify that it shouldn't be used anyways, I don't see a path where it can be invalid

Reply:
then why do we call `resources_dict_.reset();`? You can try to walk through different scenarios and try to figure out if there are any where it might be wrong or you can guarantee it won't